src/audio_stream.rs

```diff
@@ -537,24 +534,30 @@ pub(crate) fn audio_stream_fmt<T: AudioStreamSafe>(
     '\n'.fmt(f)
 }
 
-#[repr(transparent)]
-struct AudioStreamHandle(*mut ffi::oboe_AudioStream);
+struct AudioStreamHandle(*mut ffi::oboe_AudioStream, *mut c_void);
```

Comment:
What about accessing to AudioStream via shared pointer here?

Reply:
Do you mean using something like `*mut shared_ptr<oboe_AudioStream>` instead of `*mut c_void` (and maybe ideally without the pointer indirection)?

I had no experience in creating bindings between Rust and C++, so I was not  sure how to pass a `shared_ptr` between them. If you simply put `std::shared_ptr<AudioStream>` in `oboe_AudioStreamBuilder_openStreamShared` at C++ side, bindgen would use an opaque `[u64; 2]` type.

Maybe that was okay, or I could wrap the `shared_ptr` in another class or something (thinking back, that would allow accessing `AudioStream` through a getter, instead of keeping two pointers in Rust side). But I opted for allocating the `shared_ptr` on the heap, and passing a type-erased pointer to Rust side. This seemed to be less error-prone for me. But if anyone has a better idea on how to do it, I can change it.

Reply:
I mean I would like to avoid storing two pointers in `AudioStreamHandle`.

Typical `shared_ptr<T>` layout in C++ looks as follows:
```
|----------------------|
|   shared_ptr sPtr1   |
|----------------------|
|Pointer to the Object |---|-------> Object
|                      |   |           (Int value 42)
|Pointer to the Ctr blk|---|---|
|----------------------|   |   |---> Ctl block
                           |   |       (Pointer to allocator, deleter, ...)
|----------------------|   |   |       (Shared reference counter)
|   shared_ptr sPtr2   |   |   |       (Weak reference counter + offset 1)
|----------------------|   |   |
|Pointer to the Object |---|   |
|                      |       |
|Pointer to the Ctr blk|-------|
|----------------------|
```

Seems opaque type that bindgen propose is that we need (actually it is a `[*mut c_void; 2]`).

To make things totally safe we may get actual pointer to stream via function which we implements on C++ side.

Reply:
I updated #49.